plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByRepresentationAndEObjectProvider.java

```diff
@@ -0,0 +1,251 @@
+                if (semanticElement.equals(semanticDiagram)) {
+                    return diagramInResource;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public List<String> getRepresentationImagePath(Map<String, Object> parameters) throws ProviderException {
+
+        EObject rootObject = (EObject) parameters.get(ProviderConstants.KEY_CONF_ROOT_OBJECT);
+        String rootPath = (String) parameters.get(ProviderConstants.KEY_PROJECT_ROOT_PATH);
+        Session session = SessionManager.INSTANCE.getSession(rootObject);
+        Object diagramDescriptionName = parameters.get(KEY_DIAGRAM_DESCRIPTION_NAME);
+        Object targetRootObject = parameters.get(KEY_TARGET_ROOT_OBJECT);
+        if (!(diagramDescriptionName instanceof String)) {
+            throw new ProviderException(
+                    "Image cannot be computed because no diagram description name has been provided to the provider \""
+                        + this.getClass().getName() + "\"");
+        } else if (!(targetRootObject instanceof EObject)) {
+            throw new ProviderException(
+                    "Image cannot be computed because no root EObject has been provided to the provider \""
+                        + this.getClass().getName() + "\"");
+        } else {
+            checkDiagramDescriptionExist(session, (String) diagramDescriptionName);
+            List<DRepresentation> representations = getAssociatedRepresentationByName((EObject) targetRootObject,
+                    (String) diagramDescriptionName, session);
+            List<String> resultList = new ArrayList<String>();
+            for (DRepresentation dRepresentation : representations) {
+                if (dRepresentation instanceof DSemanticDiagram) {
+                    DSemanticDiagram dsd = (DSemanticDiagram) dRepresentation;
+                    String filePath = getDiagramImageFilename(dsd, rootPath);
+                    File file = new File(filePath);
+                    file.getParentFile().mkdirs();
+                    final IPath path = new Path(filePath);
+                    final Diagram gmfDiagram = getGmfDiagram(dsd);
+
+                    // Use the DiagramEditPartService to use the figure validation
+                    // infinite loop safe ViewpointDiagramGraphicalViewer.
+                    final CopyToImageUtil imageUtility = new DiagramEditPartService();
+
+                    final EditingDomain editingDomain = session.getTransactionalEditingDomain();
+                    final Diagram realOne = (Diagram) editingDomain.getResourceSet()
+                            .getEObject(EcoreUtil.getURI(gmfDiagram), true);
+                    try {
+                        imageUtility.copyToImage(realOne, path, ImageFileFormat.JPEG, new NullProgressMonitor(),
+                                PreferencesHint.USE_DEFAULTS);
+                        resultList.add(filePath);
```

Comment:
The loop body should be extracted in a method and, probably factored with the other provider.